src/calendar/__utils.ts

```diff
@@ -40,7 +40,7 @@ export function generateDaysInMonthArray(
       const daysInWeek = [...new Array(7).keys()].reduce(
         (days: Date[], dayIndex) => {
           const day = weekIndex * 7 + dayIndex - monthStartsAt + 1;
-          const cellDate = new Date(year, month, day);
+          const cellDate = new Date(year, month, day, new Date().getHours());
```

Comment:
Dont want to do this.
We can do the calculation ourselves when returning the value onChange

Reply:
We should have proper internal state rather than manipulating the output when doing onChange.